### DISCUSSION TITLE
HTML5/WebAssembly support

### DISCUSSION THREAD

**Discussion by LukasBanana** [LukasBanana/LLGL#132] [General]
I hope @vstanchevici doesn't mind me cherry picking his contribution as I just merged `dev-wasm` branch with a8907d6. This adds WebAssembly (Wasm) support via Emscripten.
As for changes from his original commits:
- Changed `Emscripten*` names to `Wasm*` since I consider WebAssembly the platform and Emscripten the toolchain to build for that platform.
- Currently only the *Animation* and *Fonts* examples work. Others need fixing in the GL backend which is now built with a third profile "WebGL".
- *BuildWasm.sh* offers a new option `--pthreads` as pthreads are disabled by default. Enabling them requires the Chrome browser to be launched with `--enable-features=SharedArrayBuffer`, which is not something an every day web user will do.

**Comment by vstanchevici** [comment]
I want to point out some important things.
The code for WebAssembly is incomplete, as it was hastily written with the primary goal of testing the library. For instance, mouse input is not fully implemented. Overall, I appreciate LLGL for its architecture, abstraction layer, and the range of supported platforms. However, I’ve decided to stop using it because WebGPU is now supported in many browsers, including Chrome, where it’s enabled by default. Consequently, I’ve chosen to transition to WebGPU instead of continuing with WebGL.

**Reply by LukasBanana** [reply]
That's understandable and I'm interested in WebGPU as well, but I'm trying to get the WebGL backend up and running first as the GL renderer serves more or less as a reference implementation for LLGL.
I also hooked up the mouse and touch input and moved the Wasm surface implementation to the `Canvas` interface (for mobile platforms) as opposed to the `Window` interface (for desktop platforms).
